lib/src/main/java/growthbook/sdk/java/GrowthBook.java

```diff
@@ -117,10 +117,16 @@ private void initializeEvalContext() {
                 new EvaluationContext.StackContext(), options);
     }
 
+    private EvaluationContext getEvaluationContext() {
+        // Reset the stackContext for every evaluation.
+        this.evaluationContext.setStack(new EvaluationContext.StackContext());
```

Comment:
I don't know how Java handles concurrency, but will this cause any race conditions if 2 evaluations are running at the same time?  Or is it always single threaded and not an issue?

Reply:
Our present `GrowthBook` Instance is not built for concurrency. If the same instance is used by multiple threads a lot of changes are needed to address the data inconsistencies and race conditions. To support the concurrent execution, we'll have to synchronize the execution and re-org the way the current evaluation is happening.